src/components/entity/ha-entity-state-picker.ts

```diff
@@ -9,9 +9,22 @@ import "../ha-combo-box";
 import type { HaComboBox } from "../ha-combo-box";
 import { formatAttributeValue } from "../../data/entity_attributes";
 import { fireEvent } from "../../common/dom/fire_event";
+import { fetchDateWS } from "../../data/history";
+import { computeDomain } from "../../common/entity/compute_domain";
 
 export type HaEntityPickerEntityFilterFunc = (entityId: HassEntity) => boolean;
 
+const DYNAMIC_STATE_DOMAINS = [
+  "sensor",
+  "input_text",
+  "text",
+  "input_number",
+  "number",
+  "person",
+  "device_tracker",
+  "zone",
+];
```

Comment:
While I do think it is cool, I'm not sure if I'm in favor of this change/addition for all the domains listed here.

It could be helpful for `text`, `input_text`, `person` and `device_tracker`.

However, `input_number`, `number` and `zone` are numerical sensors, they should be used with a numeric state condition/trigger.

For `sensor`, this is not a correct solution, as in those cases the `enum` device class should be used instead (which will provide translations as well).

Reply:
Understood, in that case I'll probably just abandon this. 

Person/device_tracker I assume could just grab the list of all zones during get_states, since I assume that list must already be available somewhere. 

That just leaves only text, which is too limited in usefulness for me to pursue this. 

I mainly wanted this as a catchall fix for the entire universe of sensors where users would not immediately know what kind of states to expect, and this would help with that discovery. If you don't like that idea, then I don't think this is worth the added effort and complexity. 

Thanks for the review.

Reply:
I agree with @frenck. We should encourage user to use sensor options because it supports translations. Also, numerical sensor can have too many states. 

But, using zone for device_tracker and person is a very good idea! 🙂

Reply:
> Person/device_tracker I assume could just grab the list of all zones during get_states, since I assume that list must already be available somewhere.

Sure, but they also are not limited to those, so this method here might still be viable.

> I mainly wanted this as a catchall fix for the entire universe of sensors where users would not immediately know what kind of states to expect, and this would help with that discovery.

Sensors already provide a construct for that (which doesn't seem to be utilized by z2m right now).

> then I don't think this is worth the added effort and complexity.

As said, I think this is still viable for `text`, `input_text`, `person` and `device_tracker` entities.